### PR TITLE
Changing return types for actions

### PR DESCRIPTION
**Pull request by devoNOTbevo**
https://github.com/bluebill1049/little-state-machine/issues/30#issuecomment-611223099

### CODE REVIEW

**Comment by devoNOTbevo** [comment]
I could only get the generic type signature to work with a variable (not an empty function that returns nothing).  I'm not sure if that's a good pattern or not - please provide feedback if not.

File: src/types.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export type Options = {
   shouldReRenderApp?: boolean;
 };
 
-export type Action = (store: any, payload: any) => any;
+export type Action = <T>(payload: T) => T;
```
Comment (bluebill1049) [inline src/types.ts:23]:
```suggestion
export type Action = <T>(store: any, payload: T) => T;
```

**Comment by bluebill1049** [comment]
awesome stuff! it's better than what we got anyhow 👍

**Comment by JonasSchulz** [comment]
Hey Guys! We ran into the same problem, when is this MR gonna be released via npm? :)

**Comment by bluebill1049** [comment]
i do it tonight @JonasSchulz

**Comment by JonasSchulz** [comment]
That is neat! Thanks!

**Comment by JonasSchulz** [comment]
Just got the new version. Fixed the problems just as it should! Thanks again :)

**Pull request merged**